napari/layers/base/base.py

```diff
@@ -215,13 +222,29 @@ def _update_coordinates(self):
         """Insert the cursor position (x, y) into the correct position in the
         tuple of indices and update the cursor coordinates.
         """
+        displayed = np.where(self.displayed)[0]
```

Comment:
personally i feel like this is much more elegant (note that `np.where(arr)` == `np.nonzero(arr)`)
```suggestion
        displayed, = np.nonzero(self.displayed)
```

but maybe what it does is less immediately obvious

Reply:
honestly this is repeated so much you should just turn it into a convenience property

Reply:
fixed all of them!